tensorflow/core/framework/resource_mgr.cc

```diff
@@ -107,14 +107,30 @@ ResourceMgr::ResourceMgr(const string& default_container)
 ResourceMgr::~ResourceMgr() { Clear(); }
 
 void ResourceMgr::Clear() {
-  mutex_lock l(mu_);
-  for (const auto& p : containers_) {
-    for (const auto& q : *p.second) {
+  // In some cases (e.g. DatasetConstructorTest.testSharedName),
```

Comment:
IMHO, comments like this one tend to become obsolete over time at which point they add little value (and potentially cause confusion).

I would simply say here that the we do the deallocation outside of the lock to avoid a potential deadlock in case any of the destructors access the resource manager.